File: src/main/java/com/damianryan/octopus/model/Page.java
package com.damianryan.octopus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import org.springframework.lang.Nullable;

import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PACKAGE)
public class Page<T> {

    long count;

    @Nullable
    String next;

    @Nullable
    String previous;

    @JsonProperty("results")
    List<T> content;
}

File: src/main/kotlin/com/damianryan/octopus/model/Page.kt
package com.damianryan.octopus.model

import com.fasterxml.jackson.annotation.JsonProperty

open class Page<T> {
    val count: Long = 0
    val next: String? = null
    val previous: String? = null
    @JsonProperty("results") val content: List<T>? = null

    override fun toString(): String {
        return "${javaClass.simpleName}(previous: $previous, next: $next, content: ${content?.size}/$count)"
    }
}